### PR TITLE
Simple mob code refactor.

### PR DESCRIPTION
**Pull request by DominikPanic**
This updates our simple mob code to /tg/ standard.
/tg/ code is not perfect, but it's 1000x better than what we currently have. Keep this in mind when making change requests.

### CODE REVIEW

File: code/game/objects/effects/spiders.dm [hunk 1]
```diff
@@ -1,213 +1,193 @@
...
+			if(!vents.len)
 				entry_vent = null
-	//=================
+				return
+			var/obj/machinery/atmospherics/components/unary/vent_pump/exit_vent = pick(vents)
+			if(prob(50))
+				visible_message("<B>[src] scrambles into the ventilation ducts!</B>", \
+								"<span class='italics'>You hear something scampering through the ventilation ducts.</span>")
+
+			spawn(rand(20,60))
```
Comment (spookydonut) [inline code/game/objects/effects/spiders.dm]: spawn? hahaha nope

File: code/game/objects/effects/spiders.dm [hunk 2]
```diff
@@ -1,213 +1,193 @@
...
+				forceMove(exit_vent)
+				var/travel_time = round(get_dist(loc, exit_vent.loc) / 2)
+				spawn(travel_time)
```
Comment (spookydonut) [inline code/game/objects/effects/spiders.dm]: again

File: code/game/objects/effects/spiders.dm [hunk 3]
```diff
@@ -1,213 +1,193 @@
...
+
+					if(!exit_vent || exit_vent.welded)
+						forceMove(entry_vent)
+						entry_vent = null
+						return
+
+					if(prob(50))
+						audible_message("<span class='italics'>You hear something scampering through the ventilation ducts.</span>")
+					sleep(travel_time)
```
Comment (spookydonut) [inline code/game/objects/effects/spiders.dm]: again

Comment (spookydonut) [reply code/game/objects/effects/spiders.dm]: use do_after

File: code/game/objects/effects/spiders.dm [hunk 4]
```diff
@@ -1,213 +1,193 @@
...
-	visible_message("<span class='warning'> [src] splits open.</span>")
+/obj/structure/spider/cocoon/Initialize()
+	icon_state = pick("cocoon1", "cocoon2", "cocoon3")
+	return ..()
+
+
+/obj/structure/spider/cocoon/Destroy()
+	var/turf/T = get_turf(src)
+	src.visible_message("<span class='warning'>\The [src] splits open.</span>")
 	for(var/atom/movable/A in contents)
```
Comment (spookydonut) [inline code/game/objects/effects/spiders.dm]: pretty sure theres a helper proc for this

File: code/game/objects/effects/spiders.dm (hunk 3 above)
Comment (DominikPanic) [reply code/game/objects/effects/spiders.dm]: do_after requires a mob

File: code/game/objects/effects/spiders.dm (hunk 4 above)
Comment (DominikPanic) [reply code/game/objects/effects/spiders.dm]: The closes I saw was on_mob_delete, do you know of one for atoms?

File: code/game/objects/effects/spiders.dm [hunk 5]
```diff
@@ -1,213 +1,201 @@
...
-						entry_vent = null
-						var/area/new_area = get_area(loc)
-						if(new_area)
-							new_area.Entered(src)
-			else
+			var/list/vents = list()
+			var/datum/pipeline/entry_vent_parent = entry_vent.parents[1]
+			for(var/obj/machinery/atmospherics/components/unary/vent_pump/temp_vent in entry_vent_parent.other_atmosmch)
+				vents.Add(temp_vent)
+			if(!vents.len)
```
Comment (Rohesie) [inline code/game/objects/effects/spiders.dm]:
```suggestion
			if(!length(vents))
```

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 6]
```diff
@@ -1,182 +1,206 @@
...
+	emote_taunt = list("growls")
+	speak_emote = list("creaks")
+	taunt_chance = 30
 
-/mob/living/simple_animal/hostile/mimic/death()
-	..()
-	qdel(src)
+	faction = list("mimic")
+	move_to_delay = 9
+	del_on_death = 1
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]:
```suggestion
	del_on_death = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 7]
```diff
@@ -1,182 +1,206 @@
...
+	if(owner != creator)
+		LoseTarget()
+		creator = owner
+		faction |= "[REF(owner)]"
+
 
-		O.loc = src
+/mob/living/simple_animal/hostile/mimic/copy/proc/CheckObject(obj/O)
+	if((isitem(O) || isstructure(O)) && !is_type_in_list(O, GLOB.protected_objects))
+		return 1
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]:
```suggestion
		return TRUE
```

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 8]
```diff
@@ -1,182 +1,206 @@
...
+	return 0
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]:
```suggestion
	return FALSE
```

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 9]
```diff
@@ -1,182 +1,206 @@
...
+
+/mob/living/simple_animal/hostile/mimic/copy/machine
+	speak = list("HUMANS ARE IMPERFECT!", "YOU SHALL BE ASSIMILATED!", "YOU ARE HARMING YOURSELF", "You have been deemed hazardous. Will you comply?", \
+				 "My logic is undeniable.", "One of us.", "FLESH IS WEAK", "THIS ISN'T WAR, THIS IS EXTERMINATION!")
+	speak_chance = 7
+
+
+/mob/living/simple_animal/hostile/mimic/copy/machine/CanAttack(atom/the_target)
+	if(the_target == creator)
+		return 0
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]:
```suggestion
		return FALSE
```

**Comment by Rohesie** [comment]
Skimmed it through. Won't bother with change requests besides a few suggestions you can merely accept via web as this is old/bad /tg/ code that's better than our older/badder one, and beats having no NPC mobs at all.
Should give more possibilities for events as well.

**Pull request merged**